lib/perl/Genome/Model/ReferenceAlignment/Command/GenotypeMicroarrayConcordance.pm

```diff
@@ -252,28 +275,99 @@ sub resolve_genotype_microarray_vcf_and_sample {
         die($self->error_message);
     }
     unless ($sort_cmd->execute) {
-        $self->error_message('Failed to sort genotype microarray VCF: '. $microarray_vcf);
+        $self->error_message('Failed to sort VCF: '. $input_vcf);
         die($self->error_message);
     }
-    return ($sorted_microarray_vcf, $genotype_sample);
+    return 1;
 }
 
-sub sorted_microarray_vcf_for_genotype_sample { 
-    my ($self, $genotype_sample) = Params::Validate::validate_pos(
-        @_, {type => OBJECT, isa => __PACKAGE__}, {type => OBJECT, isa => 'Genome::Sample'},
-    );
+sub create_temp_file_path {
+    my $self = shift;
+    my $basename = shift;
 
-    # Check if we have seen this sample, and return the sorted vcf file name
-    my $sorted_microarray_vcf = $self->_genotype_samples_and_sorted_vcfs->{$genotype_sample->id};
-    return $sorted_microarray_vcf if $sorted_microarray_vcf;
+    if ($self->output_dir) {
+        return File::Spec->join($self->output_dir,$basename);
+    } else {
+        return Genome::Sys->create_temp_file_path($basename);
+    }
+}
 
-    # Create and store the sorted VCF file name for this sample
-    $sorted_microarray_vcf = Genome::Sys->create_temp_file_path(
-        Genome::Utility::Text::sanitize_string_for_filesystem($genotype_sample->name).'_microarray_sorted.vcf'
+sub resolve_builds {
+    my $self = shift;
+
+    my @builds;
+    if ($self->lane_qc_builds) {
+        my @instrument_data = $self->build->instrument_data;
```

Comment:
This lookup logic better belongs on the build's class, that way any other commands that would like to look up lane_qc_builds associated with this build could do so without replicating this function.  Its nearly impossible to find this kind of functionality if its in these commands, so it gets replicated over and over throughout the system.